src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,414 @@
+            }
+
+            if record[_TYPE] == _WITHDRAWAL or record[_TYPE] == _DEPOSIT:
+                is_deposit = record[_TYPE] == _DEPOSIT
+                is_withdrawal = record[_TYPE] == _WITHDRAWAL
+
+                # Intra
+                kwargs.update(
+                    {
+                        'unique_id': Keyword.UNKNOWN.value,
+                        'from_exchange': self.__EXCHANGE_NAME if is_withdrawal else Keyword.UNKNOWN.value,
+                        'from_holder': self.account_holder if is_withdrawal else Keyword.UNKNOWN.value,
+                        'to_exchange': self.__EXCHANGE_NAME if is_deposit else Keyword.UNKNOWN.value,
+                        'to_holder': self.account_holder if is_deposit else Keyword.UNKNOWN.value,
+                        'crypto_sent': Keyword.UNKNOWN.value if is_deposit else str(amount),
+                        'crypto_received': str(amount) if is_deposit else Keyword.UNKNOWN.value,
+                    }
+                )
+                result.append(IntraTransaction(**kwargs))
+                continue
+
+            kwargs.update({
+                'exchange': self.__EXCHANGE_NAME,
+                'holder': self.account_holder,
+                'transaction_type': Keyword.BUY.value if float(record[_AMOUNT]) > 0 else Keyword.SELL.value,
+                'crypto_fee': '0' if is_fiat_asset else record[_FEE],
+                'fiat_fee': record[_FEE] if is_fiat_asset else None,
+            })
+
+            if record[_TYPE] == _TRADE and 'USD' not in record[_ASSET]:
+                kwargs.update({
+                    'spot_price': trade_history[record[_REFID]][_PRICE],
+                })
+                if float(record[_AMOUNT]) > 0:
+                    kwargs.update({
+                        'crypto_in': str(amount),
+                        'fiat_in_no_fee': str(float(trade_history[record[_REFID]][_COST]) - float(
+                            trade_history[record[_REFID]][_FEE])),
+                        'fiat_in_with_fee': trade_history[record[_REFID]][_COST],
+                    })
+                    result.append(InTransaction(**kwargs))
+                else:
+                    kwargs.update({
+                        'crypto_out_no_fee': str(amount),
+                        'crypto_out_with_fee': str(amount + float(record[_FEE])),
+                        'fiat_out_no_fee': str(float(trade_history[record[_REFID]][_COST]) - float(
+                            trade_history[record[_REFID]][_FEE])),
+                        'is_spot_price_from_web': False,
+                    })
+                    result.append(OutTransaction(**kwargs))
+            elif record[_TYPE] == _MARGIN or record[_TYPE] == _ROLLOVER:
+                kwargs.update({
+                    'transaction_type': Keyword.SELL.value,
+                    'crypto_out_no_fee': str(amount),
+                    'crypto_out_with_fee': str(amount + float(record[_FEE])),
```

Comment:
RP2Decimal

Reply:
I should make things RP2Decimal? The input for In/Out/Intra require string input. Am I missing something?

Reply:
No, just don't do float operations and don't convert to float. Here's what I mean:
```suggestion
                    'crypto_out_with_fee': str(amount + RP2Decimal(record[_FEE])),
```

Reply:
Ok Done. Just curious why is float frowned upon over `Decimal` or in this case `RP2Decimal`?

Reply:
It's because float is not precise enough for our calculations and so we use RP2Decimal, a subclass of Decimal that implements high-precision math and is used throughout our code.